### ISSUE TITLE
Conda env create error

### ISSUE THREAD

**Issue by DexterLagan** [Vision-CAIR/MiniGPT-4#8]
MacBook Air M2

Steps to reproduce:

> conda env create -f environment.yml 
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound: 
  - cudatoolkit

**Comment by iHaagcom** [comment]
Unfortunately, I tried creating a requirments.txt file for windows to avoid conda and I got the same error.

**Comment by DexterLagan** [comment]
I'll find out how to install cudatoolkit on conda. Vicuna runs perfect on my M2 24GB, so I'm hoping MiniGPT will also run. I can't see the visual encoder taking that much more RAM, considering Vicuna runs on just 8GB of RAM or less. Cheers

**Comment by DexterLagan** [comment]
Was able to install nearly all packages into the conda env by chmodding 666 conda's environments.txt, and removing cudatoolkit from the environment.yaml. There should be a way to run this model on CPU only.

**Comment by Arnold1** [comment]
@DexterLagan are you able to run Vicuna and MiniGPT on m2 already? is it able to use the gpu on m2?

**Comment by DexterLagan** [comment]
@Arnold1 Yes I run most models on the M2 using LLama.cpp and Alpaca.cpp. I believe they run on the GPU when using LLama.cpp. Vicuña runs very fast and well. Alpaca 7B runs faster. Alpaca 30B runs slowly, but surely. My MacBook Air is 24GB, most models fit just fine. In fact I've had more success on Mac than on Windows to run models locally - since most repos focus on Linux and Mac somehow.

**Comment by Arnold1** [comment]
@DexterLagan how you run it - what comands?

**Comment by DexterLagan** [comment]
Follow the instructions on the LLama.cpp repo for Mac. The goal was to run on Mac in the first place. Cheers !

**Comment by Coca-Cola1999** [comment]
[I want to run it in MacbookPro M1,but have a ValueError: Tokenizer class LLaMATokenizer does not exist or is not currently imported.    After executing this command : python -m fastchat.model.apply_delta --base llama-7b-hf/  --target vicuna-7b/  --delta vicuna-7b-delta-v1.1/.  Help me please！]  my email：a18622563703@163.com

**Comment by DexterLagan** [comment]
@Coca-Cola1999 Make sure you search issues on the LLama.cpp repo, and post one if nothing matches. I haven't encountered this error. Cheers!

**Comment by huma0605** [comment]
@DexterLagan thanks so much for sharing your experience! Could you also share how you connect Vicuna (running through Llama.cpp) and this MiniGPT4? I am running Llama.cpp Vicuna on my MacBook now, but still trying to figure out how to connect it to MiniGPT4. Some directions would be appreciated!

**Comment by gino8080** [comment]
can we use docker to run it on Silicon ?

**Comment by wacdev** [comment]
<img width="1554" alt="image" src="https://user-images.githubusercontent.com/127726508/235336985-91ff851d-6ec1-4e37-bfbf-4802130bb14c.png">

see pull reqeust https://github.com/Vision-CAIR/MiniGPT-4/pull/174
fix some error , now can run on mac m2 with 64GB memory

**Comment by CoruNethron** [comment]
@wacdev Thank you for fixes. Tested on M1 16Gb, with `max_new_tokens=150, max_length=1000` and all other yours changes. It can be executed. Performs very slow, because of excessive swap usage. Description of one image took 25 minutes in my case.
![Screenshot 2023-05-02 at 14 55 02](https://user-images.githubusercontent.com/4215824/236259094-514443af-6dd4-4263-b803-a345379dcc0c.png)

Trying to figure out, if `mps` device can be used to utilize GPU.

**Comment by simongcc** [comment]
@CoruNethron I have just tried to run in M1 64, not sure what is missing, after "Loading LLAMA",  it stated the error 
```
CUDA SETUP: Required library version not found: libsbitsandbytes_cpu.so. Maybe you need to compile it from source?
...
```

Is there anything missed during the installation? I am using conda created an environment dedicated for testing minigpt.

**Comment by CoruNethron** [comment]
@simongcc please check this fork https://github.com/wacfork/MiniGPT-4/tree/main by @wacdev , there was few changes, like device type (if CUDA device is not available). As far as I remember, I just took all the changes from that fork and also sightly changed `max_new_tokens` and `max_length` cause of low RAM in my system.

**Comment by simongcc** [comment]
@CoruNethron Thank you so much for shedding light on this! I will check it out and try.🙏🏽😃
